tests/common_test.py

```diff
@@ -0,0 +1,277 @@
+    host = "2001:0db8:85a3:0000:0000:8a2e:0370:7334"
+
+    prepared_host = common.prepare_host(host)
+
+    assert prepared_host == f"[{host}]"
+
+
+def testPrepareHost_whenIPv4_returnHost() -> None:
+    """Unit test for prepare_host, case when host is IPv4"""
+    host = "8.8.8.8"
+
+    prepared_host = common.prepare_host(host)
+
+    assert prepared_host == host
+
+
+def testPrepareHost_whenDomain_returnHost() -> None:
+    """Unit test for prepare_host, case when host is domain"""
+    host = "example.com"
+
+    prepared_host = common.prepare_host(host)
+
+    assert prepared_host == host
+
+
+def testPrepareHost_whenLink_returnHost() -> None:
+    """Unit test for prepare_host, case when host is link"""
+    host = "https://example.com"
+
+    prepared_host = common.prepare_host(host)
+
+    assert prepared_host == host
+
+
+def testBuildVulnLocation_whenMatchedAtIsIpv4_returnsVulnLocation() -> None:
+    """Ensure that when matched_at is an IPv4, BuildVulnLocation returns a valid VulnLocation."""
+    matched_at = "70.70.70.70:443"
+
+    vuln_location = common.build_vuln_location(matched_at)
+
+    assert vuln_location is not None
+    ipv4_asset = vuln_location.asset
+    assert isinstance(ipv4_asset, ipv4.IPv4)
+    assert ipv4_asset.host == "70.70.70.70"
+    assert ipv4_asset.version == 4
+    assert ipv4_asset.mask == "32"
+    assert len(vuln_location.metadata) == 1
+    assert (
+        any(
+            metadata.metadata_type.name == "PORT" and metadata.value == "443"
+            for metadata in vuln_location.metadata
+        )
+        is True
+    )
+
+
+def testBuildVulnLocation_whenMatchedAtIsIpv6_returnsVulnLocation() -> None:
+    """Ensure that when matched_at is an IPv6, BuildVulnLocation returns a valid VulnLocation."""
+    matched_at = "2001:0db8:85a3:0000:0000:8a2e:0370:7334"
+
+    vuln_location = common.build_vuln_location(matched_at)
+
+    assert vuln_location is not None
+    ipv6_asset = vuln_location.asset
+    assert isinstance(ipv6_asset, ipv6.IPv6)
+    assert ipv6_asset.host == "2001:0db8:85a3:0000:0000:8a2e:0370:7334"
+    assert ipv6_asset.version == 6
+    assert ipv6_asset.mask == "128"
+    assert len(vuln_location.metadata) == 0
+
+
+def testBuildVulnLocation_whenMatchedAtIsDomain_returnsVulnLocation() -> None:
+    """Ensure that when matched_at is a domain, BuildVulnLocation returns a valid VulnLocation."""
+    matched_at = "https://www.google.com"
+
+    vuln_location = common.build_vuln_location(matched_at)
+
+    assert vuln_location is not None
+    domain_asset = vuln_location.asset
+    assert isinstance(domain_asset, domain_name.DomainName)
+    assert domain_asset.name == "www.google.com"
+    assert len(vuln_location.metadata) == 1
+    assert (
+        any(
+            metadata.metadata_type.name == "URL" and metadata.value == matched_at
+            for metadata in vuln_location.metadata
+        )
+        is True
+    )
```

Comment:
Please assert on all the metadata in all the test to make sure the whole values are met.

Reply:
Updated check tests/virus_total_agent_test.py